### PR TITLE
Add non-breaking space character to EmojiPlugin message split regex

### LINKED ISSUE #9064: Emoji plugin does not work with special spaces
**Describe the bug**
Emoji's don't appear correctly if the space between emojis is typed with AltGr+space.

**To Reproduce**
1. Hold down AltGr
2. Type `@@@ @@@ @@@`
3. Release AltGr
4. Press Enter
5. Observe the :tada: emoji not showing

**Expected behavior**
Emoji shows as it should, like above should show :tada: :tada: :tada: 

**Screenshots**
image

**Environment:**
 - OS: Linux (cc reported seeing this too, I expect they are on Windows)
 - RuneLite version: 1.5.26
 - Launcher version: 1.6.2

### PR DESCRIPTION
**Pull request by WilliamCollishaw**
Resolves #9064

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/emojis/EmojiPlugin.java [hunk 1]
```diff
@@ -164,7 +164,7 @@ public void onOverheadTextChanged(final OverheadTextChanged event)
 	@Nullable
 	private String updateMessage(final String message)
 	{
-		final String[] messageWords = message.split(" ");
+		final String[] messageWords = message.split("[\\s\\u00A0]");
```
Comment (abextm) [inline runelite-client/src/main/java/net/runelite/client/plugins/emojis/EmojiPlugin.java]: this should be precompiled

**Comment by WilliamCollishaw** [comment]
> Please add some unit tests to confirm this is working. LGTM otherwise.

Added unit tests

File: runelite-client/src/main/java/net/runelite/client/plugins/emojis/EmojiPlugin.java (hunk 1 above)
Comment (WilliamCollishaw) [reply runelite-client/src/main/java/net/runelite/client/plugins/emojis/EmojiPlugin.java]: @abextm done.

File: runelite-client/src/test/java/net/runelite/client/plugins/emojis/EmojiPluginTest.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2017, Adam <Adam@sigterm.info>
```
Comment (abextm) [inline runelite-client/src/test/java/net/runelite/client/plugins/emojis/EmojiPluginTest.java]: I'm reasonably sure it is currently 2019, and your name is not Adam

File: runelite-client/src/main/java/net/runelite/client/plugins/emojis/EmojiPlugin.java [hunk 3]
```diff
@@ -26,8 +26,11 @@
 
 import java.awt.image.BufferedImage;
 import java.util.Arrays;
+import java.util.regex.Pattern;
 import javax.annotation.Nullable;
 import javax.inject.Inject;
+
```
Comment (abextm) [inline runelite-client/src/main/java/net/runelite/client/plugins/emojis/EmojiPlugin.java]: remove this newline

File: runelite-client/src/test/java/net/runelite/client/plugins/emojis/EmojiPluginTest.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+ */
+package net.runelite.client.plugins.emojis;
+
+import com.google.inject.Guice;
+import com.google.inject.testing.fieldbinder.Bind;
+import com.google.inject.testing.fieldbinder.BoundFieldModule;
+import net.runelite.api.Client;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
```
Comment (abextm) [inline runelite-client/src/test/java/net/runelite/client/plugins/emojis/EmojiPluginTest.java]: remove this newline

File: runelite-client/src/test/java/net/runelite/client/plugins/emojis/EmojiPluginTest.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+import net.runelite.client.config.ChatColorConfig;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
```
Comment (abextm) [inline runelite-client/src/test/java/net/runelite/client/plugins/emojis/EmojiPluginTest.java]: remove this newline

File: runelite-client/src/main/java/net/runelite/client/plugins/emojis/EmojiPlugin.java (hunk 3 above)
Comment (WilliamCollishaw) [reply runelite-client/src/main/java/net/runelite/client/plugins/emojis/EmojiPlugin.java]: Done

File: runelite-client/src/test/java/net/runelite/client/plugins/emojis/EmojiPluginTest.java (hunk 4 above)
Comment (WilliamCollishaw) [reply runelite-client/src/test/java/net/runelite/client/plugins/emojis/EmojiPluginTest.java]: Done

File: runelite-client/src/test/java/net/runelite/client/plugins/emojis/EmojiPluginTest.java (hunk 5 above)
Comment (WilliamCollishaw) [reply runelite-client/src/test/java/net/runelite/client/plugins/emojis/EmojiPluginTest.java]: Done

**Comment by Adam-** [comment]
I don't think this is actually a problem. Who chats with nbsp?

**Comment by WilliamCollishaw** [comment]
@Adam-, let me know if you want this fix, then i'll resolve the conflict :)

**Comment by Unmoon** [comment]
My client for whatever reason writes nbsp with AltGr + space, which i often hit when typing @@@, as per the issue.

**Comment by deathbeam** [comment]
This should be fixed imo, just because we handle nbsp in same way everywhere else, so why not here?

**Comment by WilliamCollishaw** [comment]
Resolved conflict and retested

File: runelite-client/src/main/java/net/runelite/client/plugins/emojis/EmojiPlugin.java [hunk 6]
```diff
@@ -166,9 +167,9 @@ public void onOverheadTextChanged(final OverheadTextChanged event)
 	}
 
 	@Nullable
-	private String updateMessage(final String message)
+	public String updateMessage(final String message)
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/emojis/EmojiPlugin.java]:
```suggestion
	String updateMessage(final String message)
```

File: runelite-client/src/test/java/net/runelite/client/plugins/emojis/EmojiPluginTest.java [hunk 7]
```diff
@@ -113,4 +115,17 @@ public void testGtLt()
 
 		verify(messageNode).setRuneLiteFormatMessage("<img=10>");
 	}
+
+	@Test
+	public void testEmojiUpdateMessage()
+	{
+		String PARTY_POPPER = "<img=" + (-1 + Emoji.getEmoji("@@@").ordinal()) + '>';
+		String OPEN_MOUTH = "<img=" + (-1 + Emoji.getEmoji(":O").ordinal()) + '>';
+		assertNull( emojiPlugin.updateMessage("@@@@@"));
```
Comment (Adam-) [inline runelite-client/src/test/java/net/runelite/client/plugins/emojis/EmojiPluginTest.java]:
```suggestion
		assertNull(emojiPlugin.updateMessage("@@@@@"));
```

File: runelite-client/src/test/java/net/runelite/client/plugins/emojis/EmojiPluginTest.java [hunk 8]
```diff
@@ -113,4 +115,17 @@ public void testGtLt()
...
+		assertEquals( PARTY_POPPER, emojiPlugin.updateMessage("@@@"));
```
Comment (Adam-) [inline runelite-client/src/test/java/net/runelite/client/plugins/emojis/EmojiPluginTest.java]:
```suggestion
		assertEquals(PARTY_POPPER, emojiPlugin.updateMessage("@@@"));
```

File: runelite-client/src/test/java/net/runelite/client/plugins/emojis/EmojiPluginTest.java [hunk 9]
```diff
@@ -113,4 +115,17 @@ public void testGtLt()
...
+		assertEquals( PARTY_POPPER + ' ' + PARTY_POPPER, emojiPlugin.updateMessage("@@@ @@@"));
```
Comment (Adam-) [inline runelite-client/src/test/java/net/runelite/client/plugins/emojis/EmojiPluginTest.java]:
```suggestion
		assertEquals(PARTY_POPPER + ' ' + PARTY_POPPER, emojiPlugin.updateMessage("@@@ @@@"));
```

File: runelite-client/src/test/java/net/runelite/client/plugins/emojis/EmojiPluginTest.java [hunk 10]
```diff
@@ -113,4 +115,17 @@ public void testGtLt()
...
+		assertEquals( PARTY_POPPER + ' ' + OPEN_MOUTH, emojiPlugin.updateMessage("@@@\u00A0:O"));
```
Comment (Adam-) [inline runelite-client/src/test/java/net/runelite/client/plugins/emojis/EmojiPluginTest.java]:
```suggestion
		assertEquals(PARTY_POPPER + ' ' + OPEN_MOUTH, emojiPlugin.updateMessage("@@@\u00A0:O"));
```

File: runelite-client/src/test/java/net/runelite/client/plugins/emojis/EmojiPluginTest.java [hunk 11]
```diff
@@ -113,4 +115,17 @@ public void testGtLt()
...
+		assertEquals( PARTY_POPPER + ' ' + OPEN_MOUTH + ' ' + PARTY_POPPER, emojiPlugin.updateMessage("@@@\u00A0:O @@@"));
```
Comment (Adam-) [inline runelite-client/src/test/java/net/runelite/client/plugins/emojis/EmojiPluginTest.java]:
```suggestion
		assertEquals(PARTY_POPPER + ' ' + OPEN_MOUTH + ' ' + PARTY_POPPER, emojiPlugin.updateMessage("@@@\u00A0:O @@@"));
```

File: runelite-client/src/test/java/net/runelite/client/plugins/emojis/EmojiPluginTest.java [hunk 12]
```diff
@@ -113,4 +115,17 @@ public void testGtLt()
...
+		assertEquals( PARTY_POPPER + " Hello World " + PARTY_POPPER, emojiPlugin.updateMessage("@@@\u00A0Hello World\u00A0@@@"));
```
Comment (Adam-) [inline runelite-client/src/test/java/net/runelite/client/plugins/emojis/EmojiPluginTest.java]:
```suggestion
		assertEquals(PARTY_POPPER + " Hello World " + PARTY_POPPER, emojiPlugin.updateMessage("@@@\u00A0Hello World\u00A0@@@"));
```

**Pull request merged**